lib/matplotlib/axes/_axes.py

```diff
@@ -1397,6 +1397,15 @@ def eventplot(self, positions, orientation='horizontal', lineoffsets=1,
         else:
             positions = [np.asanyarray(positions)]
 
+        if 'snap' not in kwargs and len(positions) > 200:
```

Comment:
I think this fix is probably fine, except I would just set snap to False unless otherwise requested.  I don't see the advantage of snapping for this visualization.